src/genie/libs/parser/iosxr/show_rib.py

```diff
@@ -0,0 +1,200 @@
+""" show_rib.py
+    supports commands:
+        * show rib table
+        * show rib summary 
+"""
+
+# Python
+import re
+# Metaparser
+from genie.metaparser import MetaParser
+from genie.metaparser.util.schemaengine import Schema, Any, Optional
+
+# parser utils
+from genie.libs.parser.utils.common import Common
+
+# =============================================
+# Parser for 'show rib table'
+# =============================================
+
+class ShowRibTablesSchema(MetaParser):
+    schema = {'rib_tables': {
+                        'table_id':{
+                            Any(): {
+                                'prefix_count': int,
+                                'prefix_limit': int,
+                                'prefix_limit_notified': str,
+                                'safi': str,
+                                'table_deleted': str,
+                                'table_id': str,
+                                'table_name': str,
+                                'table_reached_convergence': str,
+                                'table_version': int,
+                                'vrf_name': str,
+                                'forward_referenced' : str,
+                        }
+                    }
+                }
+            }  
+
+class ShowRibTables(ShowRibTablesSchema):
+    """ Parser for show rib tables"""
+
+    
+    cli_command = ['show rib tables']
+
+    def cli(self,port="",output=None):
+        """parsing mechanism: cli
+        Function cli() defines the cli type output parsing mechanism which
+        typically contains 3 steps: exe
+        cuting, transforming, returning
+        """
+        if output is None:
+            out = self.device.execute(self.cli_command)
```

Comment:
This will work if line 44 is not a list!

Reply:
Good spot, corrected.